packages/compiler/formatter/print/printer.ts

```diff
@@ -1002,22 +999,23 @@ export function printModelProperty(
   ];
 }
 
-function isStringSafeToUnquote(id: StringLiteralNode, options: CadlPrettierOptions): boolean {
-  const unquotedRawText = getRawText(id, options).slice(1, -1);
```

Comment:
This used to handle not trying to prettify escaped strings. So `{ "\n": string }` would not change on format, but now we will try to put an actual newline between strings. It should be possible to do the same sort of check as before against the raw text unbackticked to fix this.

Reply:
Thanks for pointing out this bug. I resolved this by escaping special characters when printing in https://github.com/microsoft/typespec/pull/1551/commits/c37318144553e352b5a9b48ec0e7042fc8247938